### PR TITLE
testcases for uncovered lines in models.py

### PR DESCRIPTION
**Pull request by Divkothuru**

### CODE REVIEW

File: tests/unit_test/api/api_processor_test.py [hunk 1]
```diff
@@ -2249,3 +2252,127 @@ def test_get_organization_not_exists(self):
...
+    def test_validate_confirm_password(cls):
+        with pytest.raises(ValueError, match="Password and Confirm Password does not match"):
+            RegisterAccount.validate_confirm_password(SecretStr("Owners@4"), {"password": SecretStr("Winte@20")})
+
+    def test_validate_password(cls):
+        with pytest.raises(ValueError, match="Password length must be 8\n"
+                                             "Missing 1 uppercase letter\nMissing 1 special letter"):
+            RegisterAccount.validate_password(SecretStr("Owners"), {})
+
+    def test_check(cls):
```
Comment (udit-pandey) [inline tests/unit_test/api/api_processor_test.py:2322]: cover other negative scenarios:
1. empty string,
2. None

File: tests/unit_test/api/api_processor_test.py [hunk 2]
```diff
@@ -2249,3 +2252,127 @@ def test_get_organization_not_exists(self):
...
+            {"name": "know_user", "type": "FORM_START"},
+            {"type": "FORM_END"},
+            {"name": "utter_submit", "type": "BOT"},
+        ]
+
+        request = StoryRequest(name="registration", type="STORY", steps=temp)
+        result = request.get_steps()
+        temp[3] = {'name': None, 'type': 'FORM_END'}
+        assert result == temp
+        assert all([i for i in temp if type(i) is dict])
```
Comment (udit-pandey) [inline tests/unit_test/api/api_processor_test.py:2378]: test case for Password class' validate_password()

File: tests/unit_test/api/api_processor_test.py [hunk 3]
```diff
@@ -2249,3 +2252,127 @@ def test_get_organization_not_exists(self):
...
+                                             "Missing 1 uppercase letter\nMissing 1 special letter"):
+            RegisterAccount.validate_password(SecretStr("Owners"), {})
+
+    def test_check(cls):
+        with pytest.raises(ValueError, match="Provide key from key vault as value"):
+            HttpActionParameters.check({"parameter_type": "key_vault", "key": "key"})
+
+    def test_validate_organization_empty(self):
+        with pytest.raises(ValueError, match="Organization can not be empty"):
+            IDPConfig.validate_organization("", {})
```
Comment (udit-pandey) [inline tests/unit_test/api/api_processor_test.py:2328]: one case missing for None value

File: tests/unit_test/api/api_processor_test.py [hunk 4]
```diff
@@ -2249,3 +2252,127 @@ def test_get_organization_not_exists(self):
...
+    def test_validate_headers_empty(cls):
+        result = EventConfig.validate_headers("", {})
+        assert result == {}
+
+    def test_validate_headers(cls):
+        result = EventConfig.validate_headers({"headers": "Headers"}, {})
+        assert result == {"headers": "Headers"}
+
+    def test_validate_config_empty(cls):
+        result = IDPConfig.validate_config("", {})
```
Comment (udit-pandey) [inline tests/unit_test/api/api_processor_test.py:2358]: None check

File: tests/unit_test/api/api_processor_test.py [hunk 5]
```diff
@@ -2249,3 +2252,127 @@ def test_get_organization_not_exists(self):
...
+            result = config.validate_ws_url(None, {})
+            assert result is None
+
+    def test_validate_ws_url(cls):
+        config = EventConfig(ws_url="ws_url", headers={"headers": "Headers"}, method="method")
+        result = config.validate_ws_url("https://www.google.com", {})
+        assert result is "https://www.google.com"
+
+    def test_validate_headers_empty(cls):
+        result = EventConfig.validate_headers("", {})
```
Comment (udit-pandey) [inline tests/unit_test/api/api_processor_test.py:2350]: None check

**Review by udit-pandey — CHANGES_REQUESTED** [review]

**Comment by Divkothuru** [comment]
changes has to be done

**Pull request closed without merging**